### PR TITLE
!find command

### LINKED ISSUE #63: "find" feature
ecmabot has `!find blah` which lists all the factoids who have "blah" in their contents. jellobot should as well.

### PR DESCRIPTION
**Pull request by brigand**
A basic !find command. There's some simple scoring of matches, which we might want to alter to produce different results. The number of results could also be increased, or pagination implemented (syntax suggestions?). 

---

Resolves #63, cc @ljharb

### CODE REVIEW

File: src/plugins/factoids/factoidsPlugin.js [hunk 1]
```diff
@@ -6,16 +6,81 @@ function parseMsg(msg) {
...
+    return `I found a matching factoid: !${first}`;
+  }
+
+  const andMore = matches.length === MATCH_LIMIT;
+  const items = matches.slice(0, DISPLAY_LIMIT).map((k) => `!${k}`);
+  const last = items.pop();
+  const tail = andMore ? `and more` : `and ${last}`;
+  const formatted = [...items, tail].join(', ');
+
+  return `I found matching factoids: ${formatted}`;
```
Comment (ljharb) [inline src/plugins/factoids/factoidsPlugin.js:63]:
```suggestion
  return `Found: ${formatted}`;
```
so the list can fit more things in it

File: src/plugins/factoids/factoidsPlugin.js [hunk 2]
```diff
@@ -6,16 +6,81 @@ function parseMsg(msg) {
...
+    .sort((a, b) => points[b] - points[a])
+    .slice(0, MATCH_LIMIT);
+
+  if (matches.length === 0) {
+    return `I couldn't find any factoid triggers/messages with that text.`;
+  }
+  if (matches.length === 1) {
+    const [first] = matches;
+    return `I found a matching factoid: !${first}`;
+  }
```
Comment (ljharb) [inline src/plugins/factoids/factoidsPlugin.js:55]: in this case, ecmabot just prints the factoid directly. why this approach?

File: src/plugins/factoids/factoidsPlugin.js [hunk 3]
```diff
@@ -6,16 +6,81 @@ function parseMsg(msg) {
...
   if (msg.from === 'ecmabot') {
     fs.writeFile('/tmp/disable-factoids', 'x', () => {});
   }
 
+  const { nick, find } = parseMsg(msg);
+
+  if (find) {
+    const response = findCommand(find);
+    if (nick) {
+      msg.respond(`${nick}, ${response}`);
```
Comment (ljharb) [inline src/plugins/factoids/factoidsPlugin.js:76]:
```suggestion
      msg.respond(`${nick}: ${response}`);
```
not sure if it matters either way

File: src/plugins/factoids/factoidsPlugin.js (hunk 1 above)
Comment (brigand) [reply src/plugins/factoids/factoidsPlugin.js]: Went with lowercase 'found' and no punctuation due to the mention.

File: src/plugins/factoids/factoidsPlugin.js (hunk 2 above)
Comment (brigand) [reply src/plugins/factoids/factoidsPlugin.js]: Added the factoid text to the response when only one match.

File: src/plugins/factoids/factoidsPlugin.js (hunk 3 above)
Comment (brigand) [reply src/plugins/factoids/factoidsPlugin.js:79]: I think we use comma for everything else, so going to keep it consistent with that.

**Pull request merged**